### DISCUSSION TITLE
Passing array to new flow

### DISCUSSION THREAD

**Discussion by SimonVillage** [directus/directus#15161] [Archive]
I have a webhook calling data from an external api endpoint. The response of this webhook is an array so `$last.data` should contain an array.

Now I am trying to trigger another flow with this array to create data in my database but I have no idea how I can pass the array?
Sending `{{$last.data}}` as payload does nothing and changed the payload to `"{{$last.data}}"`.

How can I pass an array to a new flow which I received from a webhook?

**Comment by azrikahar** [comment]
~~Can you try `$last.payload.data` instead? Or if the `data` is indeed the root object from the response itself, then it would be `$last.payload.data.data`.~~

Scratch that. Seems like `{{$last.data}}` does in fact contain the response data. How are you passing the array currently? Can you share a quick screenshot of the create data operation?

**Reply by mighty-hq** [reply]
So what's the answer exactly?

**Reply by paescuj** [reply]
I think the answer here is to add it as a raw value (using `Edit Raw Value` on the field) as outlined by @benosman in https://github.com/directus/directus/discussions/15161#discussioncomment-4208036.

**Comment by chaoticefx** [comment]
Can I ask what the solution to this was? 

I am facing a similar situation where I would like to use Trigger Flow to execute multiple flows based on the array output of Get Rows. Currently if I pass the data into the second flow via the template the data clearly goes through:

```
{
  "user": "{{ $accountability.user }}",
  "data": "{{ $last }}"
}
```

However, if I attempt to use {{ $last }} as an input to the Payload argument for Trigger Flow so that the Payload becomes an array, it adds additional quotes around the input as {{ $last }} is not valid JSON. Is there another way for me to pass in {{ $last }} as Trigger Flow's Payload that I might be missing?

**Comment by benosman** [comment]
@chaoticefx 

I ran into the same issue, and it left me perplexed for a while.

The solution/workaround I found was to access the `Edit Raw Value` option for the payload.

Hover over the field label `Payload` until the down caret shows, click that to reveal the menu. Choose `Edit Raw Value` and set the value `"{{ $last }}"` (or whatever you need) including the quotes. 

Save the operation and then try to reopen and if it works it should not add the escaping backslashes to the string and you should be able to go in and edit raw value again. If you forget and edit the payload directly it will likely add the quotes again.

This could certainly do with some UX polish to make it a bit more intuitive.

**Comment by mighty-hq** [comment]
> I have a webhook calling data from an external api endpoint. The response of this webhook is an array so `$last.data` should contain an array.
> 
> Now I am trying to trigger another flow with this array to create data in my database but I have no idea how I can pass the array? Sending `{{$last.data}}` as payload does nothing and changed the payload to `"{{$last.data}}"`.
> 
> How can I pass an array to a new flow which I received from a webhook?

So the triggerred payload looks like this

{
"data": [
{
"id": "a84b069c-43b8-4734-9f46-3b79e4ffeff6",
"sort": null,
"email": "[rzvngu5@gmail.com](mailto:rzvngu5@gmail.com)",
....................

When I try to create data using the data from the trigger with the following:

{
"email": "{{ $last.data.email }}",
}

The email is undefined.

{
"payload": {
"email": "undefined",
"full_name": "undefined",

...

What am I doing wrong? Tried a couple of variations.

**Reply by paescuj** [reply]
Could you share some some more details? How does your flow look like, which triggers and settings are you using? Perhaps a screenshot would already bring some clarity.

**Reply by omarkhouly35** [reply]
has this been answered because i am facing the same problem

**Comment by sharkfin009** [comment]
going absolutely batty with this lol. Low code hell! But i'm sure it will be improved as the rest of directus is absolutely fantastic
